pkg/loop/internal/relayer/pluginprovider/chainreader/test/chain_reader_loop_tester.go

```diff
@@ -33,11 +33,11 @@ func TestAllEncodings(t *testing.T, test func(chainreader.EncodingVersion) func(
 
 type LoopTesterOpt func(*contractReaderLoopTester)
 
-// WrapChainReaderTesterForLoop allows you to test a [types.ContractReader] implementation behind a LOOP server
-func WrapChainReaderTesterForLoop(wrapped ChainReaderInterfaceTester[*testing.T], opts ...LoopTesterOpt) ChainReaderInterfaceTester[*testing.T] {
+// WrapChainComponentsTesterForLoop allows you to test a [types.ContractReader] and [types.ChainWriter] implementation behind a LOOP server
+func WrapChainComponentsTesterForLoop(wrapped ChainComponentsInterfaceTester[*testing.T], opts ...LoopTesterOpt) ChainComponentsInterfaceTester[*testing.T] {
```

Comment:
This should be called `WrapContractReaderTesterForLoop` since it only wraps contract reader server